From 625ff6f45689aea88b0a13e3bf54686ae52bcaa5 Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Thu, 27 Feb 2020 16:28:49 +0100
Subject: [PATCH 05/17] MediaSession: remove onDirectIndividualizationRequest()

This callback is no longer used in ce_cdm v15.
Also this commit updates some trace messages to use
TRACE_L1 macro rather than printf.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
Change-Id: I4660a84a179d6e735e7b58eedd22371fee826153
---
 MediaSession.cpp | 13 +++++++------
 MediaSession.h   |  1 -
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/MediaSession.cpp b/MediaSession.cpp
index 760fe2c..d77d77c 100644
--- a/MediaSession.cpp
+++ b/MediaSession.cpp
@@ -44,7 +44,10 @@ MediaKeySession::MediaKeySession(widevine::Cdm *cdm, int32_t licenseType)
     , m_initDataType(widevine::Cdm::kCenc)
     , m_licenseType((widevine::Cdm::SessionType)licenseType)
     , m_sessionId("") {
-  m_cdm->createSession(m_licenseType, &m_sessionId);
+
+  if (widevine::Cdm::kSuccess != m_cdm->createSession(m_licenseType, &m_sessionId)) {
+    TRACE_L1("%s:%d createSession error\n",__FUNCTION__, __LINE__);
+  }
 
   ::memset(m_IV, 0 , sizeof(m_IV));;
 }
@@ -60,7 +63,7 @@ void MediaKeySession::Run(const IMediaKeySessionCallback *f_piMediaKeySessionCal
 
     widevine::Cdm::Status status = m_cdm->generateRequest(m_sessionId, m_initDataType, m_initData);
     if (widevine::Cdm::kSuccess != status) {
-       printf("generateRequest failed\n");
+       TRACE_L1("%s: generateRequest() failed\n", __FUNCTION__);
        m_piCallback->OnKeyMessage((const uint8_t *) "", 0, "");
     }
   }
@@ -90,7 +93,7 @@ void MediaKeySession::onMessage(widevine::Cdm::MessageType f_messageType, const
     break;
   }
   default:
-    printf("unsupported message type %d\n", f_messageType);
+    TRACE_L1("unsupported message type %d\n", f_messageType);
     break;
   }
   message.append(f_message.c_str(),  f_message.size());
@@ -166,6 +169,7 @@ void MediaKeySession::onKeyStatusError(widevine::Cdm::Status status) {
     errorStatus = "UnExpectedError";
     break;
   }
+  TRACE_L1("%s: errorStatus=%s", __FUNCTION__, errorStatus.c_str());
   m_piCallback->OnError(0, CDMi_S_FALSE, errorStatus.c_str());
 }
 
@@ -186,9 +190,6 @@ void MediaKeySession::onRemoveComplete() {
 void MediaKeySession::onDeferredComplete(widevine::Cdm::Status) {
 }
 
-void MediaKeySession::onDirectIndividualizationRequest(const string&) {
-}
-
 CDMi_RESULT MediaKeySession::Load(void) {
   CDMi_RESULT ret = CDMi_S_FALSE;
   g_lock.Lock();
diff --git a/MediaSession.h b/MediaSession.h
index 8448e94..98238d2 100644
--- a/MediaSession.h
+++ b/MediaSession.h
@@ -84,7 +84,6 @@ public:
     void onKeyStatusChange();
     void onRemoveComplete();
     void onDeferredComplete(widevine::Cdm::Status);
-    void onDirectIndividualizationRequest(const std::string&);
 
 private:
     void onKeyStatusError(widevine::Cdm::Status status);
-- 
2.17.1

